### PR TITLE
use tag instead of branch name for jazzy-release.

### PR DESCRIPTION
**Pull request by fujitatomoya**
address https://github.com/ros2/ros2/issues/1558

### CODE REVIEW

**Review by thomasmoore-torc — APPROVED** [review]

File: ros2.repos [hunk 1]
```diff
@@ -30,23 +30,23 @@ repositories:
...
-    version: 1.3.1
+    version: v1.3.1
   eclipse-cyclonedds/cyclonedds:
     type: git
     url: https://github.com/eclipse-cyclonedds/cyclonedds.git
     version: 0.10.4
   eclipse-iceoryx/iceoryx:
     type: git
     url: https://github.com/eclipse-iceoryx/iceoryx.git
-    version: 2.0.5
```
Comment (fujitatomoya) [inline ros2.repos:49]: there is no such tag or branch, so main branch was used???

Comment (fujitatomoya) [reply ros2.repos:49]:
```bash
=== src/eProsima/foonathan_memory_vendor (git) ===
Could not checkout ref '1.3.1': fatal: invalid reference: 1.3.1
...
=== src/eclipse-iceoryx/iceoryx (git) ===
Could not checkout ref '2.0.5': fatal: invalid reference: 2.0.5
```

`eProsima/foonathan_memory_vendor` uses master branch and, `eclipse-iceoryx/iceoryx` uses main branch.

Comment (thomasmoore-torc) [reply ros2.repos:49]: Looking at the [checks](https://github.com/ros2/ros2/actions/runs/8886395407/job/24399754571?pr=1555) for the [PR](https://github.com/ros2/ros2/pull/1555) where this version was set, we see the following output:

> Found git repository 'https://github.com/eclipse-iceoryx/iceoryx.git' but unable to verify non-branch / non-tag ref '2.0.5' without cloning the repo

It seems as if the "Validate repositories" check isn't particularly useful in this scenario.

It's also interesting that the PR mentions that `vcs export --export-with-tags src > ros2.repos` was used to generate the `ros2.repos` file but it seemed to not export all of the branches as tags. Could there be an issue with the `vcs` command or maybe there's a procedural issue associated with creating the `ros2.repos` file that needs to be addressed to ensure that valid tags (and not branches) end up in the file?

**Review by marcoag — APPROVED** [review]

**Review by ahcorde — APPROVED** [review]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/21016/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/15319/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/21706/)

**Review by clalancette — APPROVED** [review]

**Comment by fujitatomoya** [comment]
@clalancette @marcoag is there anything i should do on my side for this PR? or just leave this to you?

regarding to test failures, i do not think those are related to this PR. (https://github.com/ros2/ros2/issues/1558#issuecomment-2099435909)

**Comment by clalancette** [comment]
> @clalancette @marcoag is there anything i should do on my side for this PR? or just leave this to you?

We'll take care of merging it.

**Pull request merged**